Library/Formula/clang-format.rb

```diff
@@ -0,0 +1,46 @@
+class ClangFormat < Formula
+  homepage "http://clang.llvm.org/docs/ClangFormat.html"
+  version "2014-12-03"
+
+  stable do
+    # The 2014-12-03 tag is not imported in the github mirror, but resolves to this SHA.
+    url "https://github.com/llvm-mirror/llvm/archive/67351a195b7f7980b5a40f06222743585c5fd1f5.tar.gz"
+    sha1 "083b0966ca15c5cfb2962f28f842111e6410eba4"
+
+    resource "libcxx" do
+      url "http://llvm.org/releases/3.5.0/libcxx-3.5.0.src.tar.xz"
+      sha1 "c98beed86ae1adf9ab7132aeae8fd3b0893ea995"
+    end
+
+    resource "clang" do
+      url "https://github.com/llvm-mirror/clang/archive/a8d0bedcffc9f45a07045a95d80b0af9b3b520f1.tar.gz"
+      sha1 "c08b7ea3949e6d01bcca171abbc0108a86329bef"
+    end
+  end
+
+  depends_on "cmake" => :build
+  depends_on "ninja" => :build
+  depends_on "subversion" => :build
+
+  # Apple's libstdc++ is too old to build LLVM
+  fails_with :gcc
+  fails_with :llvm
+
+  def install
+    # Apple's libstdc++ is too old to build LLVM
+    ENV.libcxx if ENV.compiler == :clang
+
+    (buildpath/"projects/libcxx").install resource("libcxx")
+    (buildpath/"tools/clang").install resource("clang")
+
+    mkdir "build" do
+      system "cmake", "..", "-G", "Ninja", *std_cmake_args
+      system "ninja", "clang-format"
+      bin.install "bin/clang-format"
```

Comment:
There is a make install in the regular make file, but it doesn't work for just clang-format by itself (installs all of LLVM), and it also requires you to use the regular configure/make process, which is slow as molasses.